profiler/internal/cmemprof/pprof.go

```diff
@@ -0,0 +1,97 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+package cmemprof
+
+import (
+	"os"
+	"runtime"
+
+	"github.com/google/pprof/profile"
+)
+
+func (c *Profile) build() *profile.Profile {
+	// TODO: can we be sure that there won't be other allocation samples
+	// ongoing that write to the sample map? Right now it's called with c.mu
+	// held but it wouldn't be good if this (probably expensive) function
+	// was holding the same lock that recordAllocationSample also wants to
+	// hold
+	p := &profile.Profile{}
+	m := &profile.Mapping{
+		ID:   1,
+		File: os.Args[0], // XXX: Is there a better way to get the executable?
+	}
+	p.PeriodType = &profile.ValueType{Type: "space", Unit: "bytes"}
+	p.Period = 1
+	p.Mapping = []*profile.Mapping{m}
+	p.SampleType = []*profile.ValueType{
```

Comment:
NIT: Maybe add a comment linking to [Merge()](https://pkg.go.dev/github.com/google/pprof/profile#Merge) here to make it clear what we need to do here in order to get correct meta data in the final profile. E.g. leaving `TimeNanos` at `0` is fine, which wasn't immediately obvious to me when reading this.

Reply:
Done, let me know if you think there should still be more documentation on the `Merge` requirements.